File: dynamic_programming/unique_paths.py
# Problem: https://leetcode.com/problems/unique-paths/
class Solution:
    def unique_path(self, m: int, n: int) -> int:
        dp = [[1 for rows in range(n)] for cols in range(m)]
        for row in range(1, m):
            for col in range(1, n):
                dp[row][col] = dp[row-1][col] + dp[row][col - 1]
        return dp[-1][-1]

if __name__ == "__main__":
    print(Solution().unique_path(3, 3))

File: dynamic_programming/unique_paths.go
package main

import "fmt"

func uniquePaths(m int, n int) int {
	dp := make([][]int, m)
	for i := range dp {
		dp[i] = make([]int, n)
	}
	for row := 0; row < m; row++ {
		for col := 0; col < n; col++ {
			dp[row][col] = 1
		}
	}
	for row := 1; row < m; row++ {
		for col := 1; col < n; col++ {
			dp[row][col] = dp[row-1][col] + dp[row][col-1]
		}
	}
	return dp[m-1][n-1]
}

func main() {
	fmt.Println(uniquePaths(3, 2))
}
